javagen/src/main/java/com/azure/autorest/template/ModelTemplate.java

```diff
@@ -515,20 +516,18 @@ private void addProperties(ClientModel model, JavaClass classBlock, JavaSettings
 
             String fieldSignature;
             if (treatAsXml(settings, model)) {
-                if (property.isXmlWrapper()) {
+                if (property.isXmlWrapper() && !settings.isStreamStyleSerialization()) {
                     String xmlWrapperClassName = getPropertyXmlWrapperClassName(property);
-
-                    String wrapperClassDefinition = xmlWrapperClassName;
-                    if (settings.isStreamStyleSerialization()) {
-                        wrapperClassDefinition = wrapperClassDefinition + " implements XmlSerializable<" + wrapperClassDefinition + ">";
-                    }
-
-                    classBlock.staticFinalClass(JavaVisibility.PackagePrivate, wrapperClassDefinition, innerClass ->
-                        addXmlWrapperClass(innerClass, property, xmlWrapperClassName, settings));
+                    classBlock.staticFinalClass(JavaVisibility.PackagePrivate, xmlWrapperClassName,
+                        innerClass -> addXmlWrapperClass(innerClass, property, xmlWrapperClassName, settings));
 
                     fieldSignature = xmlWrapperClassName + " " + propertyName;
                 } else if (propertyType instanceof ListType) {
-                    fieldSignature = propertyType + " " + propertyName + " = new ArrayList<>()";
+                    if (settings.isStreamStyleSerialization()) {
+                        fieldSignature = propertyType + " " + propertyName + " = new LinkedList<>()";
```

Comment:
Any reason why LinkedList is used?

LinkedList may not be very cache friendly when having large number of items. (usually not a major concern though)

Reply:
I'll be reverting this, for some reason I got `LinkedList` vs `ArrayList` confused with `LinkedHashMap` vs `HashMap` where the latter doesn't preserve deserialization order.